### PR TITLE
[TechDocs] Accept sequence node custom tags in mkdocs config validation

### LINKED ISSUE #10695: [TechDocs] `!ENV` tag is considered invalid during mkdocs config validation
## Expected Behavior

The `!ENV` tag, supported by `mkdocs` since version 1.2 ([PR](https://github.com/mkdocs/mkdocs/pull/2267)), allows reading configuration parameters from environmental variables in the mkdocs config. 

As this is supported by mkdocs, it should be considered valid in the scope the config file validation.

## Actual Behavior

In `plugin-techdocs-node`s `validateMkdocsYaml` function `js-yaml` is used to validate the mkdocs config file. If it contains the `!ENV` tag, it fails the `load` operation and thus the generation step with the error message: `Failed to build the docs page: unknown tag !<!ENV>`.

## Steps to Reproduce

1. Use the `!ENV` tag in a `mkdocs.yaml` which is included in your catalog and has a valid TechDocs setup associated (`techdocs-ref` annotation, `mkdocs.yaml`, `docs/` folder and at least an `docs/index.md` file). 
2. Spin up backstage and use the Web Front to navigate to the component's TechDoc page.

## Context

We want to use the `!ENV` tag to dynamically change plugin config parameters, depending on the environment we're running our backstage instance in (local, stage, prod etc.).

## Your Environment

- `plugin-techdocs-backend@1.0.0`
- `plugin-techdocs-node@1.0.0`
- `plugin-techdocs-backend@1.0.0`
- `plugin-techdocs@1.0.0`
- `js-yaml@4.1.0`

### PR DESCRIPTION
**Pull request by viktorbahr**
## Hey, I just made a Pull Request!

This PR fixes #10695, making it possible to not only use custom tags with scalar node type, but also with sequence node type (such as the `!ENV` tag) in the mkdocs config.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Tests for new functionality and regression tests for bug fixes
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by iamEAP — COMMENTED** [review]
Thanks for the detailed issue and PR, @viktorbahr!  Use-case makes a lot of sense.  I'd not seen this capability in `mkdocs` before, and it gives me ideas for some improvements we can do in our internal systems too 😬 

I'm good with these changes conceptually!  Just a couple of minor things to take care of to get it in.  Comments below.

File: plugins/techdocs-node/src/stages/generate/__fixtures__/mkdocs_with_env_tag.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+site_name: !ENV [SITE_NAME, OTHER_NAME, 'My default site name']
```
[the remark this answers is not in the document]
Comment (viktorbahr) [reply plugins/techdocs-node/src/stages/generate/__fixtures__/mkdocs_with_env_tag.yml:1]: Will do :ok_hand:

File: .changeset/nice-boats-wonder.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
[the remark this answers is not in the document]
Comment (viktorbahr) [reply .changeset/nice-boats-wonder.md:1]: Sure, will change that.

**Comment by viktorbahr** [comment]
> Thanks for the detailed issue and PR, @viktorbahr! Use-case makes a lot of sense. I'd not seen this capability in `mkdocs` before, and it gives me ideas for some improvements we can do in our internal systems too grimacing
> 
> I'm good with these changes conceptually! Just a couple of minor things to take care of to get it in. Comments below.

Glad to hear that you find my contribution useful @iamEAP and thanks for your feedback - I've addressed the points mentioned by you.

**Review by iamEAP — APPROVED** [review]
💯 Thanks again @viktorbahr.

**Pull request merged**